src/common/scratch_buffer.hpp

```diff
@@ -1,18 +1,24 @@
-// scratch_buffer.h
-
-#include <inttypes.h>
-
+///
+/// \file
 /// Single place for definition of large RAM space for various tasks.
-/// Currently, there's no locking mechanism so before you use it, make sure
-/// there's nobody else who uses it.
 ///
-/// This is used by:
-/// - PNG decoder
-/// - QR code generator
+/// The scratch buffer isn't safe-to-use outside of the FreeRTOS context.
 ///
 /// TODO: use it for
 /// - SHA256 encoding
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <inttypes.h>
+
+#define SCRATCH_BUFFER_SIZE 49152
 
-#define SCRATCH_BUFFER_SIZE 49152 ///< ugly but constexpr int does not work
+uint8_t *scratch_buffer_acquire();
+void scratch_buffer_release(uint8_t *buffer);
```

Comment:
Parameter buffer is unused.
If it would be used I would suggest to use pointer to fixed size array or reference to fixed size array (if it would be C++ header)

Reply:
Fixed by removing all changes to the scratch_buffer. They weren't need in the last version this PR.